compiler/rustc_codegen_llvm/src/declare.rs

```diff
@@ -90,6 +90,28 @@ impl<'ll, 'tcx> CodegenCx<'ll, 'tcx> {
         declare_raw_fn(self, name, llvm::CCallConv, unnamed, visibility, fn_type)
     }
 
+    /// Declare an entry Function
+    ///
+    /// The ABI of this function can change depending on the target (although for now the same as
+    /// `declare_cfn`)
+    ///
+    /// If there’s a value with the same name already declared, the function will
+    /// update the declaration and return existing Value instead.
+    pub fn declare_entry_fn(
+        &self,
+        name: &str,
+        callconv: llvm::CallConv,
+        unnamed: llvm::UnnamedAddr,
+        fn_type: &'ll Type,
+    ) -> &'ll Value {
+        let visibility = if self.tcx.sess.target.default_hidden_visibility {
+            llvm::Visibility::Hidden
```

Comment:
Does it make sense for the entry function to ever be hidden? It also seems a bit weird for the `default_hidden_visiblility` field to control this since (as far as I know) there's no way for the user to override this.

Reply:
Well, I have kept it there since this function is a copy of `declare_cfn` (which was originally used to create the entry function).

Reply:
@wesleywiser After reading a bit on [visibility-styles](https://llvm.org/docs/LangRef.html#visibility-styles), I think we can safely always used `llvm::Visibility::Default` for entry function. Should I make the change?

Reply:
I think what you have is fine and since `declare_cfn` was doing the same thing, I don't see a strong reason to change it at this time.